src/commonMain/kotlin/exchange.dydx.abacus/state/manager/StateManagerAdaptor.kt

```diff
@@ -2398,29 +2417,93 @@ open class StateManagerAdaptor(
         }
     }
 
-    open fun complianceCheck() {
+    open fun handleComplianceResponse(response: String?, httpCode: Int, callback: ((ComplianceStatus) -> Unit)?) {
```

Comment:
why are these functions open?

Reply:
It's also not clear to me why this function needs a callback? Can't we just return normally?

Reply:
>why are these functions open?
Don't think it needs to be open 🤔 I was following the pattern for other screen functions here.

>It's also not clear to me why this function needs a callback? Can't we just return normally?
I used callback because I want to call the geoblock endpoint after getting the response from the screen endpoint. Let me know if there is a better way to do this!